src/core/coap/coap.cpp

```diff
@@ -672,20 +692,32 @@ void Coap::ProcessReceivedRequest(Header &aHeader, Message &aMessage, const Ip6:
         if (strcmp(resource->mUriPath, uriPath) == 0)
         {
             resource->HandleRequest(aHeader, aMessage, aMessageInfo);
+            error = kThreadError_None;
             ExitNow();
         }
     }
 
     if (mDefaultHandler)
     {
         mDefaultHandler(mDefaultHandlerContext, &aHeader, &aMessage, &aMessageInfo);
+        error = kThreadError_None;
```

Comment:
Thanks. In my opinion,
1. The fallback default handler is request handler, it should take on full responsibility in generating responses just like other handlers.
2. `Not Found` means the dispatcher fails to find a handler for the request. So I think it is not that reasonable to raise `Not Found` error in the dispatcher when the default handler is found.

Thoughts?